sql/parser/builtins.go

```diff
@@ -72,6 +74,55 @@ var builtins = map[string]builtin{
 	"upper": stringBuiltin(func(s string) (Datum, error) {
 		return DString(strings.ToUpper(s)), nil
 	}),
+
+	"substr": {
```

Comment:
both postgres and mysql have `substr` as an alias for `substring`. We should do the same.